notebooks/Distance.ipynb

```diff
@@ -0,0 +1,800 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# NetworKit Distance Tutorial"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "NetworKit provides several graph traversal and pathfinding algorithms within the `distance` module. This notebook covers most of these algorithms, and shows how to use them.\n",
+    "\n",
+    "All classes in the `distance` module implement a `run()` function that must be called after the constructor has been initialized."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import networkit as nk"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For this tutorial we will use the same graph, and the same source and target node."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Read a graph\n",
+    "G = nk.readGraph(\"../input/foodweb-baydry.konect\", nk.Format.KONECT)\n",
+    "GDir = G\n",
```

Comment:
We are keeping the directing graph because the runtime error `RuntimeError: Error, AllSimplePaths class has been implemented for directed graphs only.` if I pass an undirected graph. Is this not expected?

Reply:
I did not remember that `AllSimplePaths` only supports directed graphs. In that case, let's keep this as it is.